src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -93,63 +112,881 @@ static ColliderQuadInit D_809AFFD8 = {
+
+    return ret;
+}
+
+void EnAm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    CollisionHeader* colHeader = NULL;
+    s32 pad;
+    EnAm* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    ActorShape_Init(&this->dyna.actor.shape, 0.0f, ActorShadow_DrawCircle, 48.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gArmosSkel, &gArmosRicochetAnim, this->jointTable, this->morphTable,
+                   14);
+    Actor_SetScale(&this->dyna.actor, 0.01f);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+    Collider_InitCylinder(globalCtx, &this->hurtCollider);
+    Collider_InitCylinder(globalCtx, &this->blockCollider);
+    Collider_SetCylinder(globalCtx, &this->hurtCollider, &this->dyna.actor, &sHurtCylinderInit);
+
+    if (this->dyna.actor.params == ARMOS_STATUE) {
+        this->dyna.actor.colChkInfo.health = 1;
+        Collider_SetCylinder(globalCtx, &this->blockCollider, &this->dyna.actor, &sHurtCylinderInit);
+        this->hurtCollider.base.ocFlags1 = (OC1_ON | OC1_NO_PUSH | OC1_TYPE_1 | OC1_TYPE_2);
+        this->blockCollider.base.ocFlags1 = (OC1_ON | OC1_NO_PUSH | OC1_TYPE_PLAYER);
+        CollisionHeader_GetVirtual(&gArmosCol, &colHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+        Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->dyna.actor, ACTORCAT_BG);
+        EnAm_SetupStatue(this);
+    } else {
+        Collider_SetCylinder(globalCtx, &this->blockCollider, &this->dyna.actor, &sBlockCylinderInit);
+        Collider_InitQuad(globalCtx, &this->hitCollider);
+        Collider_SetQuad(globalCtx, &this->hitCollider, &this->dyna.actor, &sQuadInit);
+        this->dyna.actor.colChkInfo.health = 1;
+        this->dyna.actor.colChkInfo.damageTable = &sDamageTable;
+        EnAm_SetupSleep(this);
+        this->unk_258 = 0;
+    }
+
+    this->dyna.actor.colChkInfo.mass = MASS_HEAVY;
+}
+
+void EnAm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnAm* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->hurtCollider);
+    Collider_DestroyCylinder(globalCtx, &this->blockCollider);
+    //! @bug Quad collider is not destroyed (though destroy doesnt really do anything anyway)
+}
+
+void EnAm_SpawnEffects(EnAm* this, GlobalContext* globalCtx) {
+    static Vec3f velocity = { 0.0f, -1.5f, 0.0f };
+    static Vec3f accel = { 0.0f, -0.2f, 0.0f };
+    s32 i;
+    Vec3f pos;
+    Color_RGBA8 primColor = { 100, 100, 100, 0 };
+    Color_RGBA8 envColor = { 40, 40, 40, 0 };
+    s32 pad;
+
+    for (i = 4; i > 0; i--) {
+        pos.x = ((Rand_ZeroOne() - 0.5f) * 65.0f) + this->dyna.actor.world.pos.x;
+        pos.y = ((Rand_ZeroOne() - 0.5f) * 10.0f) + (this->dyna.actor.world.pos.y + 40.0f);
+        pos.z = ((Rand_ZeroOne() - 0.5f) * 65.0f) + this->dyna.actor.world.pos.z;
```

Comment:
Can these be swapped to have the `world` position come first and the random offset second?